app/forms/withdrawal_form.rb

```diff
@@ -15,7 +15,7 @@ def save!
     if valid?
       assign_attributes_to_trainee
       Trainees::Update.call(trainee: trainee, update_dqt: false)
-      Trainees::Withdraw.call(trainee:)
+      Trainees::Withdraw.call(trainee:) unless trainee.withdrawn?
```

Comment:
this call to the withdraw job happens before we actually change the trainee's state with `trainee.withdraw!`